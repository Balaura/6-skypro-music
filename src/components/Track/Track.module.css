.item {
  /* padding: 2px 0; */
  margin-bottom: 12px;
  /* border-bottom: 1px solid rgba(255, 255, 255, 0.1); */
}

.track {
     display: flex;
     flex-direction: row;
     justify-content: space-between;
     align-items: center;
   }
   
   .title {
     display: flex;
     flex-direction: row;
     align-items: center;
     width: 447px;
   }
   
   .titleImage {
     width: 51px;
     height: 51px;
     padding: 16px;
     background: #313131;
     display: flex;
     align-items: center;
     justify-content: center;
     margin-right: 17px;
   }
   
   .titleSvg {
     width: 18px;
     height: 17px;
     fill: transparent;
     stroke: #4e4e4e;
   }
   
   .titleLink {
     font-style: normal;
     font-weight: 400;
     font-size: 16px;
     line-height: 24px;
     color: #ffffff;
   }
   
   .titleSpan {
     font-style: normal;
     font-weight: 400;
     font-size: 16px;
     line-height: 24px;
     color: #4e4e4e;
   }
   
   .author {
     width: 321px;
     display: flex;
     justify-content: flex-start;
   }
   
   .authorLink {
     font-style: normal;
     font-weight: 400;
     font-size: 16px;
     line-height: 24px;
     color: #ffffff;
     text-align: left;
   }
   
   .album {
     width: 245px;
   }
   
   .albumLink {
     font-style: normal;
     font-weight: 400;
     font-size: 16px;
     line-height: 24px;
     color: #696969;
   }
   
   .time {
     display: flex;
     align-items: center;
   }
   
   .timeSvg {
     width: 14px;
     height: 12px;
     margin-right: 17px;
     fill: transparent;
     stroke: #696969;
   }
   
   .timeText {
     font-style: normal;
     font-weight: 400;
     font-size: 16px;
     line-height: 24px;
     text-align: right;
     color: #696969;
   }

   @media (max-width: 768px) {
    .track {
      flex-wrap: wrap;
    }
  
    .title, .author, .album, .time {
      width: 100%;
      margin-bottom: 5px;
    }
  }
  .playingDot {
    width: 16px;
    height: 16px;
    background-color: #b672ff;
    border-radius: 8px;
    display: block;
    animation: bubble_out 0.6s ease-in-out infinite both;
  }
  
  @keyframes bubble_out {
    0%, to {
      transform: scale(0.5);
    }
    50% {
      transform: scale(1);
    }
  }