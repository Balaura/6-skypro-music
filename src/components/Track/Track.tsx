import React from 'react';
import styles from './Track.module.css';
import { Track as TrackType } from '@/hooks/useFetchTracks';

const Track: React.FC<TrackType> = ({ title, artist, album, release_date }) => {
  const releaseYear = release_date ? new Date(release_date).getFullYear() : 'N/A';

  return (
    <div className={styles.item}>
      <div className={styles.track}>
        <div className={styles.title}>
          <div className={styles.titleImage}>
            <svg className={styles.titleSvg}>
              <use xlinkHref="img/icon/sprite.svg#icon-note"></use>
            </svg>
          </div>
          <div className={styles.titleText}>
            <a className={styles.titleLink} href="#">
              {title || 'Неизвестный трек'} <span className={styles.titleSpan}></span>
            </a>
          </div>
        </div>
        <div className={styles.author}>
          <a className={styles.authorLink} href="#">
            {artist || 'Неизвестный исполнитель'}
          </a>
        </div>
        <div className={styles.album}>
          <a className={styles.albumLink} href="#">
            {album || 'Неизвестный альбом'}
          </a>
        </div>
        <div className={styles.time}>
          <svg className={styles.timeSvg}>
            <use xlinkHref="img/icon/sprite.svg#icon-like"></use>
          </svg>
          <span className={styles.timeText}>{releaseYear}</span>
        </div>
      </div>
    </div>
  );
};

export default Track;