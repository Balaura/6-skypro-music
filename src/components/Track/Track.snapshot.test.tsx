import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Track from './Track';

const mockStore = configureStore([]);

describe('Track component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      audioPlayer: {
        isPlaying: false,
        currentTrack: null,
        favoriteTracks: [],
      },
      auth: {
        username: 'testUser',
      },
    });

    const track = {
      _id: 1,
      name: 'Test Track',
      author: 'Test Author',
      release_date: '2023-01-01',
      genre: 'Test Genre',
      duration_in_seconds: 180,
      album: 'Test Album',
      track_file: 'http://example.com/test.mp3',
      artist: 'Test Artist',
    };

    const tree = renderer.create(
      <Provider store={store}>
        <Track
          {...track}
          isPlaying={false}
          isCurrentTrack={false}
          onPlay={() => {}}
        />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});