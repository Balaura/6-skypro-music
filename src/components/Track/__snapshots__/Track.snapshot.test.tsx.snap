// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component renders correctly 1`] = `
<div
  className="item"
>
  <div
    className="track"
  >
    <div
      className="title"
    >
      <div
        className="titleImage"
        data-testid="play-button"
        onClick={[Function]}
      >
        <svg
          className="titleSvg"
        >
          <use
            xlinkHref="/img/icon/sprite.svg#icon-note"
          />
        </svg>
      </div>
      <div
        className="titleText"
      >
        <a
          className="titleLink"
          onClick={[Function]}
        >
          Test Track
          <span
            className="titleSpan"
          />
        </a>
      </div>
    </div>
    <div
      className="author"
    >
      <a
        className="authorLink"
        href="#"
      >
        Test Author
      </a>
    </div>
    <div
      className="album"
    >
      <a
        className="albumLink"
        href="#"
      >
        Test Album
      </a>
    </div>
    <div
      className="time"
    >
      <button
        className="likeBtn "
        disabled={false}
        onClick={[Function]}
      >
        <svg
          className="likeSvg"
        >
          <use
            xlinkHref="/img/icon/sprite.svg#icon-dislike"
          />
        </svg>
      </button>
      <span
        className="timeText"
      >
        3:00
      </span>
    </div>
  </div>
</div>
`;
